icu4c/source/i18n/unitconverter.cpp

```diff
@@ -450,51 +453,22 @@ std::vector<std::unique_ptr<MeasureUnit>> extractTargets(MeasureUnit source, UEr
         tempTargetUnit.dimensionality = singleUnit.getDimensionality(status);
```

Comment:
I can't convince myself that the logic in this function is correct.

It seems that you're splitting the source unit into single units (that seems fine), then getting the target unit for each of those single units (that seems fine).  What I don't follow is what happens next.

You call `TempSingleUnit::forMeasureUnit`, which assumes that there's only one single unit in the target unit, setting a U_ILLEGAL_ARGUMENT_ERROR otherwise, but the target unit could be compound.

Then you overwrite the SI prefix and dimensionality with the ones from the source single unit, but what happens if the target itself has an SI prefix, like kilogram?

Try to find some test cases where the target units are non-compound or have kilograms, and debug this code.  I highly doubt that it works at all as currently written, unless I'm missing something.

Reply:
Aha .... okay, I thought TempSingleUnit::forMeasureUnit accepts compound unit.

Reply:
I will add the test cases after implementing the loading data part.

Reply:
OK.  For now, please insert

```
// TODO(ICU-20568): Add more test coverage for this function.
```

Reply:
Done!